tests/js/common/shell/shell-statistics.js

```diff
@@ -0,0 +1,130 @@
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function CommonStatisticsSuite() {
+  'use strict';
+  let c;
+  return {
+    setUp: function () {
+      c = db._create("shellCommonStatTestCollection");
+    },
+    tearDown: function () {
+      db._drop(c);
+    },
+
+    testServerStatsStructure: function () {
+      let stats = internal.serverStatistics();
+      assertTrue(Number.isInteger(stats.transactions.started));
+      assertTrue(Number.isInteger(stats.transactions.committed));
+      assertTrue(Number.isInteger(stats.transactions.aborted));
+      assertTrue(Number.isInteger(stats.transactions.intermediateCommits));
+    },
+
+    testServerStatsCommit: function () {
+      let stats1 = internal.serverStatistics();
+      c.insert({ "gondel" : "ulf" });
+      let stats2 = internal.serverStatistics();
+
+      assertTrue(stats1.transactions.started < stats2.transactions.started);
+      assertTrue(stats1.transactions.committed < stats2.transactions.committed);
+    },
+
+    testServerStatsAbort: function () {
+      let stats1 = internal.serverStatistics();
+      let stats2;
+      try {
+        db._executeTransaction({
+          collections: {},
+          action: function () {
+            var err = new Error('abort on purpose');
+            throw err;
+          }
+        });
+        fail();
+      } catch (err) {
+        stats2 = internal.serverStatistics();
+        assertMatch(/abort on purpose/, err.errorMessage);
+      }
+
+      assertTrue(stats1.transactions.started < stats2.transactions.started);
+      assertTrue(stats1.transactions.aborted < stats2.transactions.aborted);
+    },
+
+    testIntermediateCommitsCommit: function () {
+      let stats1 = internal.serverStatistics();
+      db._query(`FOR i IN 1..3 INSERT { "ulf" : i } IN ${c.name()}`, {}, { "intermediateCommitCount" : 2});
+      let stats2 = internal.serverStatistics();
+
+      if(db._engine().name === "rocksdb") {
+        assertTrue(stats1.transactions.intermediateCommits < stats2.transactions.intermediateCommits);
+      } else {
+        assertEqual(stats1.transactions.intermediateCommits, 0);
+      }
+      assertTrue(stats1.transactions.committed < stats2.transactions.committed);
+    },
+
+    testIntermediateCommitsAbort: function () {
+      let stats1 = internal.serverStatistics();
+      let stats2;
+      try {
+        let rv = db._query(`FOR i IN 1..3
+                            FILTER ASSERT(i == 0, "abort on purpose")
+                            INSERT { "ulf" : i } IN ${c.name()}
+                           `, {}, { "intermediateCommitCount" : 2});
+        fail()
+      } catch (err) {
+        stats2 = internal.serverStatistics();
+        assertMatch(/abort on purpose/, err.errorMessage);
+      }
+
+      if(db._engine().name === "rocksdb") {
+        assertEqual(stats1.transactions.intermediateCommits, stats2.transactions.intermediateCommits);
```

Comment:
this is not necessarily safe, as some background operation (e.g. the TTL index) could have triggered an intermediate commit in the meantime.